### PR TITLE
Setting idp in mdc

### PR DESCRIPTION
**Pull request by TayGov**
# Description

This PR includes the following proposed change(s):

Adding identity provider to mdc

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/GenerateUrlTest.java [hunk 1]
```diff
@@ -48,6 +48,7 @@
 
     private static final String CODE = "CODE";
     private static final String CLIENT_APP_NAME = "clientAppName";
+    private static final String IDENTITY_PROVIDER = "test";
```
Comment (alexjoybc) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/submissionApiDelegateImpl/GenerateUrlTest.java:51]:
```suggestion
    private static final String IDENTITY_PROVIDER = "identity_provider_alias";
```

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/utils/SecurityUtils.java [hunk 2]
```diff
@@ -39,6 +39,15 @@ public static String getClientId() {
         }
     }
 
+    public static String getIdentityProvider() {
+        try {
+            return ((KeycloakPrincipal) SecurityContextHolder.getContext().getAuthentication().getPrincipal())
```
Comment (alexjoybc) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/utils/SecurityUtils.java:44]: I feel like we have a duplicate here
Can we have a private method `getCustomClaimValue(String key)` ?

**Review by alexjoybc — CHANGES_REQUESTED** [review]
Nice, a quick comment and the real value of the claim key

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/utils/SecurityUtils.java (hunk 2 above)
Comment (TayGov) [reply src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/utils/SecurityUtils.java]: Yes it should

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/utils/SecurityUtils.java [hunk 3]
```diff
@@ -30,10 +20,22 @@ public static String getClientId() {
         }
     }
 
+    public static Optional<String> getUniversalIdFromContext() {
```
Comment (alexjoybc) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/utils/SecurityUtils.java:23]: Nice!!!!!!!!!!!

**Review by alexjoybc — APPROVED** [review]
Looks awesome!

**Pull request merged**